catalyst/core/callbacks/optimizer.py

```diff
@@ -173,10 +168,19 @@ def on_batch_end(self, state: _State):
                 grad_clip_fn=self.grad_clip_fn
             )
 
-            if self.save_model_grads:
-                for tag, value in model.named_parameters():
-                    tag = tag.replace(".", "/")
-                    state.model_grads[tag] = value.grad.cpu().numpy()
+            if hasattr(state, "model_grads"):
+                memo = set()
```

Comment:
I dont like the idea of using `hasattr ` logic with State in Callbacks.
It looks like undocumented trick. Too many magic.

Reply:
Okay. Previously I've found that when I suggested adding model_grads to RunnerState ( https://github.com/catalyst-team/catalyst/pull/581/commits/5b99adeca617d8f0b74703649d6c6b377f06e08b ) 
the changes wasn't merged. So right know it's still remains undocumented feature. I could add them again in this PR and makes everything more convenient